pulp_ansible/tests/functional/cli/test_collection_upload.py

```diff
@@ -69,3 +72,57 @@ def test_upload_collection(self):
         repo = self.repo_api.read(repo.pulp_href)
         repo_version = self.repo_versions_api.read(repo.latest_version_href)
         self.assertEqual(repo_version.number, 1)  # We uploaded 1 collection
+
+    def test_upload_collection_with_requires_ansible(self):
+        """Test whether ansible-galaxy can upload a Collection to Pulp."""
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        # Create a distribution.
+        body = gen_distribution()
+        body["repository"] = repo.pulp_href
+        distribution_create = self.distributions_api.create(body)
+        created_resources = monitor_task(distribution_create.task).created_resources
+        distribution = self.distributions_api.read(created_resources[0])
+
+        self.addCleanup(self.distributions_api.delete, distribution.pulp_href)
+
+        raw_content = "https://raw.githubusercontent.com/ansible-collections/"
+        runtime_url = raw_content + "cisco.nxos/1.4.0/meta/runtime.yml"
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            collection_name = "".join([random.choice(string.ascii_lowercase) for i in range(26)])
+            cmd = "ansible-galaxy collection init --init-path {} pulp.{}".format(
+                temp_dir, collection_name
+            )
+            subprocess.run(cmd.split())
+
+            runtime_path = "{}{}/meta".format(temp_dir, "/pulp/" + collection_name)
+            cmd = "mkdir -p {}".format(runtime_path)
+            subprocess.run(cmd.split())
+            with open(f"{runtime_path}/runtime.yml", "wb") as runtime_yml:
+                runtime_yml.write(http_get(runtime_url))
+
+            cmd = "ansible-galaxy collection build --output-path {} {}{}".format(
+                temp_dir, temp_dir, "/pulp/" + collection_name + "/"
+            )
+            subprocess.run(cmd.split())
```

Comment:
Is building this once and checking the tarball as an asset in the tests area an option? If that tarball was checked in all of this code could be removed which would simplify things overall.